src/main/java/com/google/sps/data/NSWGovAdaptor.java

```diff
@@ -0,0 +1,131 @@
+package com.google.sps.data;
+
+import java.io.BufferedReader;
+import java.io.DataOutputStream;
+import java.io.InputStreamReader;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import com.google.gson.Gson;
+import com.google.gson.reflect.TypeToken;
+
+/** File to run the GET request in java before putting it in the main adaptor function. 
+ */
+public class NSWGovAdaptor {
+
+  private HashMap<Integer, GovSiteDetails> dataMap;
+
+  public NSWGovAdaptor() throws Exception {
+    dataMap = new HashMap<>();
+    this.updateSiteInfo();
+  }
+  
+  public void setMap(HashMap<Integer, GovSiteDetails> dataMap) {
```

Comment:
The map should be treated as an internal implementation detail of the class (hence, it being private). What use case is there for allowing external objects to fetch and replace the map?

Reply:
Nope, good point - I wrote it together with the getter as a get and set pair to make good habits but it is true that we will not be modifying it from the outside hence no need for the set function.